### PR TITLE
Test/add benchmark calls to tests

### PR DESCRIPTION
**Pull request by mattyg**
Currently on CI, tests are passing but some benchmarks are failing inconsistently.

This adds test for all the wasm calls that are used in the benchmarks, to clarify that the errors only arise when the calls are run in benchmarks (i.e. when the wasm is instantiated and called 1000 - 1000000 times).

Edit: clearly the benchmark failures are inconsistent since CI is now passing here.

This PR is merging into chore/cleanup-make-private for a more legible diff.

### CODE REVIEW

File: test/src/test.rs [hunk 1]
```diff
@@ -320,8 +320,8 @@ pub mod tests {
                 )
             }
         });
-        assert!(matches!(call_1.join(), Ok(_)));
-        assert!(matches!(call_2.join(), Ok(_)));
+        assert!(call_1.join().is_ok());
+        assert!(call_2.join().is_ok());
     }
```
Comment (mattyg) [inline test/src/test.rs:325]: this and above changes are unrelated clippy fixes

File: test/src/test.rs [hunk 2]
```diff
@@ -239,7 +239,7 @@ pub mod tests {
             &mut store.lock().as_store_mut(),
```
Comment (mattyg) [inline test/src/test.rs:239]: changes below are unrelated clippy fixes

File: test/src/test.rs [hunk 3]
```diff
@@ -535,4 +535,147 @@ pub mod tests {
 
         assert_eq!(result, Vec::<u8>::from([1]));
     }
+
```
Comment (mattyg) [inline test/src/test.rs:538]: The tests below are copied from the benchmark calls to the `TestWasm::Io` wasm.

**Review by ThetaSinner — APPROVED** [review]
